tests/Gedmo/Tree/NestedTreeRootRepositoryTest.php

```diff
@@ -238,8 +240,36 @@ public function testShouldRemoveRootNodeFromTree(): void
         static::assertNull($node->getParent());
     }
 
+    /**
+     * @dataProvider invalidStringMethods
+     *
+     * @param mixed $stringMethod
+     */
+    public function testGetPathAsStringWithInvalidStringMethod($stringMethod): void
+    {
+        /** @var NestedTreeRepository $repo */
+        $repo = $this->em->getRepository(self::CATEGORY);
+        $carrots = $repo->findOneBy(['title' => 'Carrots']);
+
+        $this->expectException(InvalidArgumentException::class);
+        $repo->getPathAsString($carrots, [
+            'stringMethod' => $stringMethod,
+        ]);
+    }
+
+    public function invalidStringMethods(): array
```

Comment:
```suggestion
    public function invalidStringMethods(): iterable
```
You can also yield these values instead of put an array in memory.